src/sinks/gcp/chronicle_unstructured.rs

```diff
@@ -0,0 +1,521 @@
+impl SinkConfig for ChronicleUnstructuredConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let creds = self.auth.build(Scope::MalachiteIngestion).await?;
+
+        let tls = TlsSettings::from_options(&self.tls)?;
+        let client = HttpClient::new(tls, cx.proxy())?;
+
+        let endpoint = self.endpoint("v2/unstructuredlogentries:batchCreate")?;
+
+        // For the healthcheck we see if we can fetch the list of available log types.
+        let healthcheck_endpoint = self.endpoint("v2/logtypes")?;
+
+        let healthcheck = build_healthcheck(client.clone(), &healthcheck_endpoint, creds.clone())?;
+        let sink = self.build_sink(client, endpoint, creds, cx)?;
+
+        Ok((sink, healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn sink_type(&self) -> &'static str {
+        NAME
+    }
+
+    fn acknowledgements(&self) -> Option<&AcknowledgementsConfig> {
+        Some(&self.acknowledgements)
+    }
+}
+
+impl ChronicleUnstructuredConfig {
+    fn build_sink(
+        &self,
+        client: HttpClient,
+        base_url: String,
+        creds: GcpAuthenticator,
+        cx: SinkContext,
+    ) -> crate::Result<VectorSink> {
+        use crate::sinks::util::service::ServiceBuilderExt;
+
+        let request = self.request.unwrap_with(&TowerRequestConfig {
+            rate_limit_num: Some(1000),
+            ..Default::default()
+        });
+
+        let batch_settings = self.batch.into_batcher_settings()?;
+
+        let partitioner = self.key_partitioner()?;
+
+        let svc = ServiceBuilder::new()
+            .settings(request, GcsRetryLogic)
+            .service(ChronicleService::new(client, base_url, creds));
+
+        let request_settings = RequestSettings::new(self)?;
+
+        let sink = GcsSink::new(cx, svc, request_settings, partitioner, batch_settings);
+
+        Ok(VectorSink::from_event_streamsink(sink))
+    }
+
+    fn key_partitioner(&self) -> crate::Result<KeyPartitioner> {
+        Ok(KeyPartitioner::new(self.log_type.clone()))
+    }
+
+    fn endpoint(&self, path: &str) -> Result<String, ChronicleError> {
+        Ok(format!(
+            "{}/{}",
+            match (&self.endpoint, self.region) {
+                (Some(endpoint), None) => endpoint.trim_end_matches("/"),
+                (None, Some(region)) => region.endpoint(),
+                _ => return Err(ChronicleError::RegionOrEndpoint),
+            },
+            path
+        ))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ChronicleRequest {
+    pub body: Bytes,
+    pub finalizers: EventFinalizers,
+    pub metadata: RequestMetadata,
+}
+
+impl Finalizable for ChronicleRequest {
+    fn take_finalizers(&mut self) -> EventFinalizers {
+        std::mem::take(&mut self.finalizers)
+    }
+}
+
+#[derive(Clone, Debug)]
+struct ChronicleEncoder {
+    customer_id: String,
+    encoder: crate::codecs::Encoder<()>,
+}
```

Comment:
Usually we have a `Transformer` in the encoders as well. You can get it by calling `self.encoding.transformer()` on the sink config. It can be used to reshape an event with the `except_fields`, `only_fields` and `timestamp_format` options.

It can be used by `self.transformer.transform(&mut event)` before calling `encoder.encode`